hilti/runtime/include/util.h

```diff
@@ -649,4 +650,122 @@ constexpr std::size_t hashCombine(std::size_t hash1, std::size_t hash2, Hashes..
         return result;
 }
 
+namespace control {
+
+template<typename Data, typename Error>
+class Reference;
+
+/// Helper class for loosely tracking liveliness of some memory.
+///
+/// Code using this would instantiate an instance of this pointing to some data.
+/// It can then pass out references to that control block by calling `Ref`.
+/// One can get values from these references as long as the original control
+/// block is live (this is checked).
+///
+/// \tparam Data the type of data this Block controls
+/// \tparam Error the type of exception a Reference should throw if it became stale
```

Comment:
Sorry, I don't want to be a pain, but that's not the style we use elsewhere. Maybe my "Doxygen-style" was misleading: what I meant is use `/** ... */` and `@xxx`; and no ` @brief` needed (which I see further below), first sentence is always the summary. That's more like "Javadoc-style" I guess? Essentially let's just try to stay consistent with our comment/doc style.

Reply:
> \brief, at least with my Doxygen version this would correctly emit the summary in the class overview (

Right, Doxygen is flexible in what it parses, I indeed meant this for consistency with our practice elsewhere.